src/V3LinkCells.cpp

```diff
@@ -699,10 +689,59 @@ class LinkCellsVisitor final : public VNVisitor {
     }
 };
 
+class LinkConfigsVisitor : public VNVisitor {
+    // STATE
+    LinkCellsCtx* const m_ctx;  // Context for linking cells
+
+    // VISITORS
+    void visit(AstConfig* nodep) override {
+        const string fullTopName = v3Global.opt.work() + '.' + v3Global.opt.topModule();
+        const bool topMatch = (fullTopName == nodep->name());
+        if (topMatch) {
+            m_ctx->topModuleNames.erase(fullTopName);
+            for (AstConfigCell* cellp = nodep->designp(); cellp;
+                 cellp = VN_AS(cellp->nextp(), ConfigCell)) {
+                m_ctx->topModuleNames.insert(cellp->name().c_str());
+            }
+        }
+        // We don't do iterateChildren here because we want to skip designp
+        for (AstNode* itemp = nodep->itemsp(); itemp; itemp = itemp->nextp()) { iterate(itemp); }
+    }
+
+    void visit(AstConfigCell* nodep) override {
+        nodep->v3warn(E_UNSUPPORTED, "Unsupported: config cell");
+        iterateChildren(nodep);
+    }
+    void visit(AstConfigRule* nodep) override {
+        nodep->v3warn(E_UNSUPPORTED, "Unsupported: config rule");
+        iterateChildren(nodep);
+    }
+    void visit(AstConfigUse* nodep) override {
+        nodep->v3warn(E_UNSUPPORTED, "Unsupported: config use");
+        iterateChildren(nodep);
+    }
+
+    void visit(AstNode* nodep) override { iterateChildren(nodep); }
+
+public:
+    // CONSTRUCTORS
+    LinkConfigsVisitor(AstNetlist* nodep, LinkCellsCtx* ctx)
+        : m_ctx{ctx} {
+        // Initialize top module from command line option
+        if (!m_ctx->topModuleNames.size() && !v3Global.opt.topModule().empty()) {
+            const string fullTopName = v3Global.opt.work() + '.' + v3Global.opt.topModule();
+            m_ctx->topModuleNames.insert(fullTopName);
+        }
```

Comment:
Doesn't seem right if config has two tops specified, add a test for that please.  Will need to turn off MULTITOP warning.

Reply:
Verilator supports multiple tops already (if two were in the command line or in one file), so hook into that.

It will give a MULTITOP watning, but just turn it off, e.g. -Wno-MULTITOP.

Reply:
Added! Worked out of box, surprisingly